app/src/main/java/ch/sdp/vibester/activity/DownloadActivity.kt

```diff
@@ -33,6 +33,7 @@ class DownloadActivity : AppCompatActivity() {
     //Companion object to indicate when the download completes.
     companion object {
         var downloadComplete = false
+        var downloadStarted = false
     }
```

Comment:
Why not creating a complete static module: DownloadStatus
that has 
- Private Constructor()
- Then Companion Object(){
      setDownloadComplete()
      setDownloadStarted()
      isDownloadComplete()
     isDownloadStarted()
- }

It is quite quick to write and easily testable like that it will improve the coverage :D

Reply:
Testing the static fields is done implicitly. They are there to simply be used in the tests, not to be tested by theirselves :/ It would otherwise be "I am downloading thus I am downloading" rather than a check if the download has started, which is implicitly tested in the currently commented tests. Those tests unfortunately take a long amount of time, thus they are currently commented. However, you can see the test "downloadMultipleSongs" checks for downloadComplete field, and implicitly checks the downloadStarted field.

Reply:
An important factor to mention would be that the PR does not add or remove any coverage. Although might be because of the small number of lines effected, the tests for those lines are already complete, just commented due to the reason mentioned above

Reply:
maybe not for the coverage, but DownloadStatus can be done for the sake of separation in general.